test/utils/distributed-state-registry-clusterSpec.js

```diff
@@ -0,0 +1,177 @@
+var DistributedStateRegistry = require( '../../src/utils/distributed-state-registry' );
+var LocalMessageConnector = require( '../mocks/local-message-connector' );
+
+var createRegistry = function ( serverName, messageConnector ) {
+	var options = {
+		serverName: serverName,
+		stateReconciliationTimeout: 10,
+		messageConnector: messageConnector
+	};
+	var result = {
+		addCallback: jasmine.createSpy( 'add' ),
+		removeCallback: jasmine.createSpy( 'remove' ),
+		registry: new DistributedStateRegistry( 'TEST_TOPIC', options )
+	}
+
+	result.registry.on( 'add', result.addCallback );
+	result.registry.on( 'remove', result.removeCallback );
+
+	return result;
+};
+
+describe( 'distributed-state-registry adds and removes names in a cluster', function(){
+	var messageConnector = new LocalMessageConnector();
+	var registryA;
+	var registryB;
+	var registryC;
+
+	it( 'creates the registries', function(){
+		registryA = createRegistry( 'server-name-a', messageConnector );
+		registryB = createRegistry( 'server-name-b', messageConnector );
+		registryC = createRegistry( 'server-name-c', messageConnector );
+		expect( messageConnector.subscribedTopics.length ).toBe( 3 );
+	});
+
+	it( 'adds an entry to registry a', function(){
+		registryA.registry.add( 'test-entry-a' );
+		expect( registryA.addCallback ).toHaveBeenCalledWith( 'test-entry-a' );
+		expect( registryB.addCallback ).toHaveBeenCalledWith( 'test-entry-a' );
+		expect( registryC.addCallback ).toHaveBeenCalledWith( 'test-entry-a' );
+		expect( registryA.registry.getAll() ).toEqual([ 'test-entry-a' ]);
+		expect( registryB.registry.getAll() ).toEqual([ 'test-entry-a' ]);
+		expect( registryC.registry.getAll() ).toEqual([ 'test-entry-a' ]);
+	});
+
+	it( 'adds an entry to registry b', function(){
+		registryB.registry.add( 'test-entry-b' );
+		expect( registryA.addCallback ).toHaveBeenCalledWith( 'test-entry-b' );
+		expect( registryB.addCallback ).toHaveBeenCalledWith( 'test-entry-b' );
+		expect( registryC.addCallback ).toHaveBeenCalledWith( 'test-entry-b' );
+		expect( registryA.registry.getAll() ).toEqual([ 'test-entry-a', 'test-entry-b' ]);
+		expect( registryB.registry.getAll() ).toEqual([ 'test-entry-a', 'test-entry-b' ]);
+		expect( registryC.registry.getAll() ).toEqual([ 'test-entry-a', 'test-entry-b' ]);
+	});
+
+	it( 'adds the same entry to registry c', function(){
+		registryA.addCallback.calls.reset();
```

Comment:
these can be pulled into a beforeEach, would save a lil bit of space

good practice to introduce though, resetting mocks.